### PR TITLE
Make more implementation details of bindings obvious private

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

File: src/cryptography/hazmat/bindings/openssl/binding.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def _ensure_ffi_initialized(cls):
                 cls._register_osrandom_engine()
 
     @classmethod
-    def init_static_locks(cls):
+    def _init_static_locks(cls):
```
Comment (jlaine) [inline src/cryptography/hazmat/bindings/openssl/binding.py]: Hm, that's an interesting change. I know of at least one project which will break but I'm happy to remove the call if it makes no sense:

https://github.com/aiortc/aiortc/blob/28cf22c2a82136784cc8ece6ca82143afdba5c86/aiortc/rtcdtlstransport.py#L40

Comment (alex) [reply src/cryptography/hazmat/bindings/openssl/binding.py]: `init_static_locks()` is called at module scope, so there's no way to import `Binding` without it already having been called, so yeah, it doesn't really make sense

**Comment by alex** [comment]
Oh what the hell, we also documented this...

**Comment by alex** [comment]
Ugh, the hell with this, let's just plan to delete it all once we're at 1.1.0 and greater.

**Pull request closed without merging**